### PR TITLE
encrypt/decrypt unit tests - across sdks

### PR DESCRIPTION
**Pull request by shawnarneygit**
Additional unit tests.  This is to compare encryption/decryption, using the output of the other sdks (i.e. kotlin, iOS).  

Please review and comment on issues or changes to the code or comments.  These did pass all unit tests.

### CODE REVIEW

File: tests/testnet.spec.js [hunk 1]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+      token_code: tokenCode,
+      memo,
+      hash,
+      offline_url: offlineUrl
+    }
+
+    const cipherContent = fioSDKBob.transactions.getCipherContent('new_funds_content', content, bobPrivateKey, alicePublicKey)
+   
+    expect(cipherContent).to.be.a('string')
+    // console.log(cipherContent)
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: remove

File: tests/testnet.spec.js [hunk 2]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+
+    expect(uncipherContent.payee_public_address).to.equal(bobPublicKey)
+
+    // same party, ensure cannot decipher
+    try {
+      const uncipherContentSameParty = fioSDKBob.transactions.getUnCipherContent('new_funds_content', cipherContent, alicePrivateKey, alicePublicKey)
+      console.log(uncipherContentSameParty)
+      expect(uncipherContentSameParty.payee_public_address).to.notequal(bobPublicKey)
+    } catch (e) {
+
```
Comment (andreyvEze) [inline tests/testnet.spec.js:746]: add `console.log(e)`

File: tests/testnet.spec.js [hunk 3]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+    }
+
+    // non party, ensure cannot decipher
+    try {
+      const uncipherContentNonParty = fioSDKBob.transactions.getUnCipherContent('new_funds_content', cipherContent, nonPartyPrivateKey, bobPublicKey)
+      console.log(uncipherContentNonParty)
+      expect(uncipherContentNonParty.payee_public_address).to.notequal(bobPublicKey)
+    } catch (e) {
+
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: add `console.log(e)`

File: tests/testnet.spec.js [hunk 4]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+    } catch (e) {
+     
+    }
+   
+    try {
+      const uncipherContentNonPartyA = fioSDKBob.transactions.getUnCipherContent('new_funds_content', cipherContent, bobPrivateKey, nonPartyPublicKey)
+      console.log(uncipherContentNonPartyA)
+      expect(uncipherContentNonPartyA.payee_public_address).to.notequal(bobPublicKey)
+    } catch (e) {
+
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: add `console.log(e)`

File: tests/testnet.spec.js [hunk 5]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+    expect(uncipherContent.payee_public_address).to.equal(bobPublicKey)
+
+    // same party, ensure cannot decipher
+    try {
+      const uncipherContentSameParty = fioSDKBob.transactions.getUnCipherContent('record_obt_data_content', cipherContent, alicePrivateKey, alicePublicKey)
+      console.log(uncipherContentSameParty)
+      expect(uncipherContentSameParty.payee_public_address).to.notequal(bobPublicKey)
+    } catch (e) {
+
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: add `console.log(e)`

File: tests/testnet.spec.js [hunk 6]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+    }
+
+    // non party, ensure cannot decipher
+    try {
+      const uncipherContentNonParty = fioSDKBob.transactions.getUnCipherContent('record_obt_data_content', cipherContent, nonPartyPrivateKey, bobPublicKey)
+      console.log(uncipherContentNonParty)
+      expect(uncipherContentNonParty.payee_public_address).to.notequal(bobPublicKey)
+    } catch (e) {
+
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: add `console.log(e)`

File: tests/testnet.spec.js [hunk 7]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+    } catch (e) {
+     
+    }
+   
+    try {
+      const uncipherContentNonPartyA = fioSDKBob.transactions.getUnCipherContent('record_obt_data_content', cipherContent, bobPrivateKey, nonPartyPublicKey)
+      console.log(uncipherContentNonPartyA)
+      expect(uncipherContentNonPartyA.payee_public_address).to.notequal(bobPublicKey)
+    } catch (e) {
+
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: add `console.log(e)`

File: tests/testnet.spec.js [hunk 8]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+    // UKPHFo0xs3GwGAXMc44QqkDtj7dcbGHjU3cJmN1qiYWADvzyd9pen2WwKn0VZtk0ZTGFXpap7Id8nZxlCMK7TjkabO85XNbhausE4ZZzx3hm25bqV2GDRHpRomsRDGAzLbFEumsm+4UNBtnOqUK3Kuo91vKjlLIV3NoF83qOSbhL8QDqV2N/yJxSu4PsiDeqhhSypZx8McaubVoUueioWA==
+
+    const uncipherContent = fioSDKBob.transactions.getUnCipherContent('new_funds_content', cipherContent, alicePrivateKey, bobPublicKey)
+
+    expect(uncipherContent.payee_public_address).to.equal(bobPublicKey)
+
+    // same party, ensure cannot decipher
+    try {
+      const uncipherContentSameParty = fioSDKBob.transactions.getUnCipherContent('new_funds_content', cipherContent, alicePrivateKey, alicePublicKey)
+      console.log(uncipherContentSameParty)
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: remove

File: tests/testnet.spec.js [hunk 9]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+      expect(uncipherContentSameParty.payee_public_address).to.notequal(bobPublicKey)
+    } catch (e) {
+
+    }
+
+    // non party, ensure cannot decipher
+    try {
+      const uncipherContentNonParty = fioSDKBob.transactions.getUnCipherContent('new_funds_content', cipherContent, nonPartyPrivateKey, bobPublicKey)
+      console.log(uncipherContentNonParty)
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: remove

File: tests/testnet.spec.js [hunk 10]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+      expect(uncipherContentNonParty.payee_public_address).to.notequal(bobPublicKey)
+    } catch (e) {
+     
+    }
+   
+    try {
+      const uncipherContentNonPartyA = fioSDKBob.transactions.getUnCipherContent('new_funds_content', cipherContent, bobPrivateKey, nonPartyPublicKey)
+      console.log(uncipherContentNonPartyA)
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: remove

File: tests/testnet.spec.js [hunk 11]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+    expect(cipherContent).to.be.a('string')
+    
+    const uncipherContent = fioSDKBob.transactions.getUnCipherContent('record_obt_data_content', cipherContent, alicePrivateKey, bobPublicKey)
+
+    expect(uncipherContent.payee_public_address).to.equal(bobPublicKey)
+
+    // same party, ensure cannot decipher
+    try {
+      const uncipherContentSameParty = fioSDKBob.transactions.getUnCipherContent('record_obt_data_content', cipherContent, alicePrivateKey, alicePublicKey)
+      console.log(uncipherContentSameParty)
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: remove

File: tests/testnet.spec.js [hunk 12]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+      expect(uncipherContentSameParty.payee_public_address).to.notequal(bobPublicKey)
+    } catch (e) {
+
+    }
+
+    // non party, ensure cannot decipher
+    try {
+      const uncipherContentNonParty = fioSDKBob.transactions.getUnCipherContent('record_obt_data_content', cipherContent, nonPartyPrivateKey, bobPublicKey)
+      console.log(uncipherContentNonParty)
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: remove

File: tests/testnet.spec.js [hunk 13]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+      expect(uncipherContentNonParty.payee_public_address).to.notequal(bobPublicKey)
+    } catch (e) {
+     
+    }
+   
+    try {
+      const uncipherContentNonPartyA = fioSDKBob.transactions.getUnCipherContent('record_obt_data_content', cipherContent, bobPrivateKey, nonPartyPublicKey)
+      console.log(uncipherContentNonPartyA)
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: remove

File: tests/testnet.spec.js [hunk 14]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+    }
+
+    const cipherContent = fioSDKBob.transactions.getCipherContent('new_funds_content', content, bobPrivateKey, alicePublicKey)
+   
+    expect(cipherContent).to.be.a('string')
+    // console.log(cipherContent)
+
+    // UKPHFo0xs3GwGAXMc44QqkDtj7dcbGHjU3cJmN1qiYWADvzyd9pen2WwKn0VZtk0ZTGFXpap7Id8nZxlCMK7TjkabO85XNbhausE4ZZzx3hm25bqV2GDRHpRomsRDGAzLbFEumsm+4UNBtnOqUK3Kuo91vKjlLIV3NoF83qOSbhL8QDqV2N/yJxSu4PsiDeqhhSypZx8McaubVoUueioWA==
+
+    const uncipherContent = fioSDKBob.transactions.getUnCipherContent('new_funds_content', cipherContent, alicePrivateKey, bobPublicKey)
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: could you create `const newFundsContentStr = 'new_funds_content'` and use it everywhere?

File: tests/testnet.spec.js [hunk 15]
```diff
@@ -694,3 +694,198 @@ describe('Record obt data, check', () => {
...
+      chain_code: chainCode,
+      token_code: tokenCode,
+      status: '',
+      obt_id: '',
+      memo: memo,
+      hash: hash,
+      offline_url: offlineUrl
+    }
+
+    const cipherContent = fioSDKBob.transactions.getCipherContent('record_obt_data_content', content, bobPrivateKey, alicePublicKey)
```
Comment (andreyvEze) [inline tests/testnet.spec.js]: the same for `const recordObtDataContent = 'record_obt_data_content'`

**Review by andreyvEze — APPROVED** [review]

**Comment by shawnarneygit** [comment]
resolved as per Andrew comments.  Unable to add console.log for try/catch... as the goal is to have the decrypt fail in those cases.

**Pull request merged**